perl/lib/Interface.pm

```diff
@@ -51,7 +52,7 @@ has config => (
   is            => 'ro',
   isa           => 'Str',
   coerce        => 1,
-  required      => 1,
+  required      => 0,
```

Comment:
This is no longer required, so that we can defer the check of whether we have `config` until we've seen whether we have `json_config`, which may define `config`.